transport/sessionStateMachine/common.ts

```diff
@@ -107,9 +107,14 @@ export interface SessionOptions {
    */
   heartbeatsUntilDead: number;
   /**
-   * Duration to wait between connection disconnect and actual session disconnect
+   * Max duration that a session can be without a connection before we consider
+   * it dead.
```

Comment:
Maybe a full on union is not worth it, but a comment is probably a good idea.

```suggestion
   * Max duration that a session can be without a connection before we consider
   * it dead. This value is ignored if {@link enableTransparentSessionReconnects} is
   * set to false.
```

Unless i'm misunderstanding something

Reply:
Talked in person, looks like I missed the fact that session grace is now a general indicator for time-to-connect instead of time-to-reconnect

Reply:
yup good idea i think the semantics of the option changed, i'll add a comment